src/functions.js

```diff
@@ -0,0 +1,83 @@
+let TodoItems = JSON.parse(localStorage.getItem('ToDoItems')) || [];
+const TodoItemsV2 = TodoItems;
+
+const list = document.getElementById('list');
+const display = (i) => {
+  const item = document.createElement('li');
+  const data = TodoItems[i];
+  item.innerHTML = `<input type="checkbox" class ="check-box ${data.index}">
+  <span contenteditable="true" for="item${data.index}">${data.description}</span>
+  <div class="dots">&#8942;</div>`;
+  list.appendChild(item);
+};
+
+class Item {
+  constructor(description) {
+    this.description = description;
+    this.completed = false;
+    this.index = TodoItems.length;
+  }
+}
+
+const storeData = () => {
+  localStorage.setItem('ToDoItems', JSON.stringify(TodoItems));
+};
+
+const remove = (num) => {
+  TodoItems.splice(num, 1);
+  storeData();
+  for (let i = 0; i < TodoItems.length; i += 1) {
+    const data = TodoItems[i];
+    data.index = i;
+  }
+  storeData();
+};
+
+const modify = () => {
+  const items = document.querySelectorAll('#list span');
+  for (let i = 0; i < items.length; i += 1) {
+    items[i].addEventListener('click', (event) => {
+      event.stopPropagation();
+      for (let i = 0; i < items.length; i += 1) {
+        items[i].parentElement.style.background = 'none';
+        const rmIcon = items[i].nextElementSibling;
+        rmIcon.innerHTML = '&#8942';
+        rmIcon.style.cursor = 'move';
+      }
+      items[i].parentElement.style.backgroundColor = 'rgb(255, 255, 181)';
+      const rmIcon = items[i].nextElementSibling;
+      rmIcon.innerHTML = '&#x1F5D1';
+      rmIcon.style.cursor = 'pointer';
+      rmIcon.addEventListener('click', () => {
+        remove(i);
+        rmIcon.parentElement.remove();
+      });
+    });
+    const data = TodoItems[i];
+    items[i].addEventListener('keydown', () => {
+      data.description = items[i].innerHTML;
+      storeData();
+    });
+  }
+  window.addEventListener('click', () => {
+    for (let i = 0; i < items.length; i += 1) {
+      items[i].parentElement.style.background = 'none';
+      const rmIcon = items[i].nextElementSibling;
+      rmIcon.innerHTML = '&#8942';
+      rmIcon.style.cursor = 'move';
+    }
+  });
+};
```

Comment:
- [ ] [REQUIRED] You have done a great job @Peter1907 , however, after modifying/editing and deleting a task it updates all remaining items' indexes, so they represent the current list order and are unique.
- [ ] [REQUIRED] Also ensure that the property `index` is set to the value of the new array length (i.e. if a 5th task is added to the list, the index of that task should equal 5). Your index currently starts from 0 instead of 1. Kindly implement the fix.
-  Take a look at the picture below: I only have two inputs 1, and 23 but the old input is showing in the LocalStorage. You need to ensure that the local storage is updated with the deleted items and edited items.
![Screenshot from 2022-08-10 23-22-42](https://user-images.githubusercontent.com/58692590/184032036-7541884e-789f-487a-be51-9c14437d2346.png)